app/controllers/concerns/course/forum/topic_controller_subscription_concern.rb

```diff
@@ -0,0 +1,37 @@
+module Course::Forum::TopicControllerSubscriptionConcern
+  extend ActiveSupport::Concern
+
+  def subscribe
+    if set_subscription_state
```

Comment:
So you need to ensure that the show and subscribe actions have the same permissions. Will you be defining the ability yourself?